examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+                        matrix_sum += -std::pow(omega, 2) *
+                                      quadrature_data.mass_coefficient[i][j];
+                        matrix_sum +=
+                          quadrature_data.stiffness_coefficient[i][j];
+                        cell_matrix(i, j) += matrix_sum * quadrature_data.JxW;
+                      }
+                    cell_rhs(i) += quadrature_data.right_hand_side[i];
+                  }
+              }
+            cell->get_dof_indices(local_dof_indices);
+            constraints.distribute_local_to_global(cell_matrix,
+                                                   cell_rhs,
+                                                   local_dof_indices,
+                                                   system_matrix,
+                                                   system_rhs);
+          }
+      }
+
+    system_matrix.compress(VectorOperation::add);
+    system_rhs.compress(VectorOperation::add);
+  }
+
+  // @sect4{ElasticWave::solve}
+
+  // This is even more simple than in step-40. We use the parallel direct solver
+  // MUMPS which requires less options than an iterative solver. The drawback is
+  // that it does not scale very well. It is not straightforward to solve the
+  // Helmholtz equation with an iterative solver. The shifted Laplacian
+  // multigrid method is a well known approach to precondition this system, but
+  // this is beyond the scope of this tutorial.
+  template <int dim>
+  void ElasticWave<dim>::solve()
+  {
+    TimerOutput::Scope              t(computing_timer, "solve");
+    LinearAlgebraPETSc::MPI::Vector completely_distributed_solution(
+      locally_owned_dofs, mpi_communicator);
+
+    SolverControl                    solver_control;
+    PETScWrappers::SparseDirectMUMPS solver(solver_control, mpi_communicator);
+    solver.solve(system_matrix, completely_distributed_solution, system_rhs);
+
+    pcout << "   Solved in " << solver_control.last_step() << " iterations."
+          << std::endl;
+    constraints.distribute(completely_distributed_solution);
+    locally_relevant_solution = completely_distributed_solution;
+  }
+
+  // @sect4{ElasticWave::set_position_vector}
+
+  // We use this function to calculate the values of the position vector.
+  template <int dim>
+  void ElasticWave<dim>::set_position_vector()
+  {
+    Point<dim> p;
+    for (unsigned int position_idx = 0;
+         position_idx < parameters.nb_probe_points;
+         ++position_idx)
+      {
+        // Because of the way the operator + and - are overloaded. To substract
+        // two points, the following has to be done:
+        // `Point_b<dim> + (-Point_a<dim>)`
+        p = (position_idx / ((double)(parameters.nb_probe_points - 1))) *
+              (parameters.probe_stop_point + (-parameters.probe_start_point)) +
+            parameters.probe_start_point;
+        position[position_idx][0] = p[0];
+        position[position_idx][1] = p[1];
```

Comment:
If you made `position` a `std::vector<Point<dim>>` as I commented on above, then this would simply be a straight assignment -- or, even better, you don't need the `p` variable at all :-)

Reply:
I'll do it after the support for `Point<dim>` in HDF5 has been merged :)